### ISSUE TITLE
Settings: Make parsing slightly more permissive

### ISSUE THREAD

**Issue by corani** [aumayr/fava#365]
Currently, when a certain option expects a boolean, but you give it a string containing that boolean, you will get an exception on the console:

```
[E 160918 13:22:00 wsgi:355] 500 POST /xyz/source/?account=&payee=&tag= (192.168.2.100) 1230.96ms
[E 160918 13:22:02 ioloop:629] Exception in callback <bound method LiveReloadHandler.poll_tasks of <class 'livereload.handlers.LiveReloadHandler'>>
    Traceback (most recent call last):
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/tornado/ioloop.py", line 1039, in _run
        return self.callback()
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/livereload-2.4.1-py3.5.egg/livereload/handlers.py", line 66, in poll_tasks
        filepath, delay = cls.watcher.examine()
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/livereload-2.4.1-py3.5.egg/livereload/watcher.py", line 73, in examine
        func and func()
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/beancount_fava-0.3.1.dev0-py3.5.egg/fava/cli.py", line 66, in <lambda>
        lambda: reload_source_files(api))
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/beancount_fava-0.3.1.dev0-py3.5.egg/fava/cli.py", line 61, in reload_source_files
        api.load_file()
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/beancount_fava-0.3.1.dev0-py3.5.egg/fava/api/__init__.py", line 143, in load_file
        self.fava_options, errors = parse_options(self.custom_entries)
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/beancount_fava-0.3.1.dev0-py3.5.egg/fava/api/fava_options.py", line 80, in parse_options
        key, value = _parse_option_entry(entry)
      File "/xyz/.pyenv/versions/3.5.1/lib/python3.5/site-packages/beancount_fava-0.3.1.dev0-py3.5.egg/fava/api/fava_options.py", line 57, in _parse_option_entry
        assert isinstance(value, bool)
    AssertionError
```

1. This error should be caught and shown in the web-interface instead.
2. It would be nice if we supply a string, to first try to convert it to the expected type (i.e. bool), before raising an error. Idem for numbers.

**Comment by blais** [comment]
No particular reason. I wanted to avoid going down the rabbit hole of creating a complex configuration scheme; I think so far the very small number of options is a testimony to that approach working well, though there are those couple of oddities. It could be define, one would have to update the grammar for it.

**Comment by yagebu** [comment]
Fair enough, don't really mind the small oddities, so I'll update our code to follow beancount's style of options.

**Comment by yagebu** [comment]
Implemented in f342dee42b8a33ffb7884e31b6cb04fcd0b508bd
